Saule/AttributeFormat.cs

```diff
@@ -0,0 +1,18 @@
+﻿namespace Saule
+{
+    /// <summary>
+    /// The format of attribute names for clients
+    /// </summary>
+    public enum AttributeFormat
```

Comment:
@joukevandermaas I started this in a separate branch for now.  [Over here](https://github.com/joukevandermaas/saule/compare/master...barsh:property-name-formatter-for-camelCase).  I'll bring it into this PR when it matures.

As you can see I created the property name converters as you suggested, but my next steps are unclear.  Can you provide some guidance on how you envision getting access to `jsonApiConfiguration.PropertyNameConverter` and making use of it during serialization and deserialization?

Reply:
@joukevandermaas my work is still in a separate branch [Over here](https://github.com/joukevandermaas/saule/compare/master...barsh:property-name-formatter-for-camelCase).  I'll bring it into this PR when it matures.

Here's the current status.  You can now enable camelCase as a configuration option, [see preliminary documentation here](https://github.com/barsh/saule/blob/property-name-formatter-for-camelCase/docs/content/4-customizing-serialization.md#using-camelcase-for-attribute-names)

There are several places where the default configuration is used instead of the configuration specified in `startup.cs`:

 https://github.com/joukevandermaas/saule/blob/1c90bf148eaf687ded1040e3a779f77f39be3ade/Saule/Http/JsonApiProcessor.cs#L28
 
https://github.com/joukevandermaas/saule/blob/f2402b89ad5dda9be75610a630c66669fcb535d9/Saule/Http/JsonApiMediaTypeFormatter.cs#L21

https://github.com/joukevandermaas/saule/blob/f10f52f5c927c997b275ee67ed98d15ac12ee742/Saule/Http/JsonApiAttribute.cs#L19

As far as I can tell the first two, don't introduce a problem, but the third one does.  Using the attribute filter `[JsonApi]` allows you to opt into JsonApi for some endpoints and not others (we should add this to the docs), but when doing so `DefaultPropertyNameConverter` is used instead of `CamelCasePropertyNameConverter`.  *Can you provide feedback on this?*

Still todo: Unit Tests (they are currently passing but tests need to be added for new functionality)

Reply:
Perhaps a static property `DefaultConfiguration` on `JsonApiAttribute`? Then in startup people can say 
```cs
JsonApiAttribute.DefaultConfiguration = new JsonApiConfiguration {
    PropertyNameConverter = new CamelCasePropertyNameConverter()
};
```

Not a very "nice" solution but I think it would be good enough? Definitely agree that needs to be documented if we settle on that.

Reply:
That worked and it actually turned out quite nicely, because we can have `ConfigureJsonApi` set the configuration on the attribute, see commit here: https://github.com/joukevandermaas/saule/commit/d9583087f22a2139f3aa048249be875c7f440a24

Now for unit tests...

Reply:
@joukevandermaas that worked and it actually turned out quite nicely, because we can have `ConfigureJsonApi` set the configuration on the attribute, see commit here: https://github.com/joukevandermaas/saule/commit/d9583087f22a2139f3aa048249be875c7f440a24

Now for unit tests...